packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -259,6 +295,43 @@ class DismissUpdateDetails {
   final double progress;
 }
 
+/// Details for [TriggerDismissCallback].
+///
+/// See also:
+///
+///   * [Dismissible.shouldTriggerDismiss], which receives this information.
+class TriggerDismissDetails {
```

Comment:
Do we need a new class? Could we add what we need to DismissUpdateDetails be used instead for this use case? 
If not, we should definitely document what the difference is between them. :)

Reply:
`DismissUpdateDetails` contains a `bool previousReached` that makes no sense for our `shouldTriggerDismiss` callback as it is only called once per dismiss gesture (onDragEnd). This was the main reason for creating a new class.

Also, I'm adding a new `bool isFling` parameter to `TriggerDismissDetails`, that indicates if the fling velocity is enough to be recognise as a fling gesture, the same way `reached` set to true when the threshold is passed. Using `shouldTriggerDismiss: (details) => details.reached || details.isFling` should be the same as returning null (default behaviour).

I do agree that those two classes have a lot in common tho 🤔

Reply:
Perhaps internal to the class we can have an enum that differentiates which scenario the details are representing? And non-applicable properties can be null then?

Reply:
I think my preference would be to have 2 distinct types, rather than requiring the developer to keep track of which fields are guaranteed to be null/non-null at the respective times.

A while ago, I learned that class inheritance can [reduce compile sizes](https://github.com/dart-lang/language/issues/3780#issuecomment-2094969955). So maybe we could have something like:

```dart
sealed class _DismissGestureDetails {
  final DismissDirection direction;
  final bool reached;
  final double progress;
}

class DismissUpdateDetails extends _DismissGestureDetails {
  final bool previousReached;
}

class TriggerDismissDetails extends _DismissGestureDetails {
  final double dragExtent;
  final double velocity;
}
```